proto/protovalidate/buf/validate/validate.proto

```diff
@@ -137,6 +137,46 @@ message MessageRules {
   // }
   // ```
   repeated Rule cel = 3;
+
+  // `oneof` is a repeated field of type MessageOneofRule that specifies a list of fields
+  // of which at most one can be present. If `required` is also specified, then exactly one
+  // of the specified fields _must_ be present.
+  //
+  // This will enforce oneof-like constraints with a few features not provided by
+  // actual Protobuf oneof declarations:
+  //   1. Repeated and map fields are allowed in this validation. In a Protobuf oneof,
+  //      only scalar fields are allowed.
+  //   2. Fields with implicit presence are allowed. In a Protobuf oneof, all member
+  //      fields have explicit presence. This means that, for the purpose of determining
+  //      how many fields are set, explicitly setting such a field to its zero value is
+  //      effectively the same as not setting it at all.
+  //   3. This will generate validation errors when unmarshalling, even from the binary
+  //      format. With a Protobuf oneof, if multiple fields are present in the serialized
+  //      form, earlier values are usually silently ignored when unmarshalling, with only
+  //      the last field being present when unmarshalling completes.
```

Comment:
I know I proposed this copy, but I found an issues in it. Here's a take two 😄 
```suggestion
  //   3. This will always generate validation errors for a message unmarshalled from
  //      serialized data that sets more than one field. With a Protobuf oneof, when
  //      multiple fields are present in the serialized form, earlier values are usually
  //      silently ignored when unmarshalling, with only the last field being set when
  //      unmarshalling completes.
```

Reply:
Follow-up: https://github.com/bufbuild/protovalidate/pull/378